lib/chef/provider/package/apt.rb

```diff
@@ -71,15 +71,11 @@ def candidate_version
         end
 
         def package_locked(name, version)
-          islocked = false
           locked = shell_out_compact_timeout!("apt-mark", "showhold")
-          locked.stdout.each_line do |line|
-            line_package = line.strip
-            if line_package == name
-              islocked = true
-            end
+          locked_packages = locked.stdout.each_line.map do |line|
+            line.strip
           end
-          islocked
+          name.reject { |n| locked_packages.include? n }.empty?
```

Comment:
nice cleanup as well

Reply:
Better clean up:

```ruby
locked_packages = locked.stdout.each_line.map(&:strip)
name.any? { |n| locked_packages.include?(n) }
```

That will stop if it finds one. Also I'm not sure what the semantics of this for situations with multi-package and partial locked input. I think you changed the logic from `any?` to `all?`